src/DynamoCoreWpf/UI/GuidedTour/GuidesValidationMethods.cs

```diff
@@ -603,25 +604,51 @@ internal static void LibraryScrollToBottom(Step stepInfo, StepUIAutomation uiAut
         internal static void HighlightPort(Step stepInfo, StepUIAutomation uiAutomationData, bool enableFunction, GuideFlow currentFlow)
         {
             CurrentExecutingStep = stepInfo;
+            CurrentExecutingStep.MainWindow.Dispatcher.Invoke(DispatcherPriority.Background, new Action(delegate { }));
             var stepMainWindow = CurrentExecutingStep.MainWindow as Window;
 
-            var byOriginNode = GuideUtilities.FindNodeByID(stepMainWindow, "a20a93da6af14deebe2df37c1662349f");
-            if (byOriginNode == null) return;
-            var itemsControlInputPort = GuideUtilities.FindChild(byOriginNode, "inputPortControl") as ItemsControl;
-            if (itemsControlInputPort == null) return;
-            var itemContainer = itemsControlInputPort.ItemContainerGenerator.ContainerFromIndex(0);
+            //Parse the parameters location in the json file
+            var highlightColor = uiAutomationData.Parameters[0] as string;
```

Comment:
I think it would be recommended to verify the size of the `Parameters` before accessing its position directly.

Reply:
Added a line for checking that is not null and the number of parameters, see the next commit:
https://github.com/DynamoDS/Dynamo/pull/12860/commits/ab9dac8beced65ebb949df6a3d59919273f40a45